File: eBarberShop/eBarberShop.Model/Rezervacija.cs
namespace eBarberShop.Model
{
    public class Rezervacija
    {
        public int RezervacijaId { get; set; }
        public DateTime Datum { get; set; }
        public DateTime Vrijeme { get; set; }
        public bool? Status { get; set; }
        public int KorisnikId { get; set; }
        public int UposlenikId { get; set; }

        public virtual Korisnici Korisnik { get; set; }
        public virtual Uposlenik Uposlenik { get; set; }

        public string? UposlenikIme => Uposlenik?.Ime;
        public string? UposlenikPrezime => Uposlenik?.Prezime;

        public virtual ICollection<RezervacijaUsluge> RezervacijaUsluge { get; set; } = new List<RezervacijaUsluge>();
    }
}

File: eBarberShop/UI/ebarbershop_desktop/lib/models/rezervacije/rezervacija.dart
import 'package:json_annotation/json_annotation.dart';

part 'rezervacija.g.dart';

@JsonSerializable()
class Rezervacija {
  int rezervacijaId;
  DateTime datum;
  DateTime vrijeme;
  bool? status;
  int korisnikId;
  int uposlenikId;
  int uslugaId;
  String? uposlenikIme;
  String? uposlenikPrezime;
  String? klijentIme;
  String? klijentPrezime;
  String? nazivUsluge;

  Rezervacija(
      this.rezervacijaId,
      this.datum,
      this.vrijeme,
      this.status,
      this.korisnikId,
      this.uposlenikId,
      this.uslugaId,
      this.uposlenikIme,
      this.uposlenikPrezime,
      this.klijentIme,
      this.klijentPrezime,
      this.nazivUsluge);

  factory Rezervacija.fromJson(Map<String, dynamic> json) =>
      _$RezervacijaFromJson(json);

  Map<String, dynamic> toJson() => _$RezervacijaToJson(this);
}
